cgogn/core/cmap/attribute.h

```diff
@@ -538,9 +400,132 @@ class Attribute : public AttributeOrbit<DATA_TRAITS, ORBIT>
 		return iterator(this, this->chunk_array_cont_->end());
 	}
 
-	virtual const std::string& get_type_name() const override
+protected:
+	ChunkArrayContainer*	chunk_array_cont_;
+	TChunkArray*			chunk_array_;
+
+private:
+	inline static ChunkArrayContainer* get_chunk_array_container_tmpl(MapData* map, Orbit orb)
 	{
-		return chunk_array_->get_type_name();
+		switch (orb) {
+			case Orbit::DART : return &(map->template get_attribute_container<Orbit::DART>());
+			case Orbit::PHI1 : return &(map->template get_attribute_container<Orbit::PHI1>());
+			case Orbit::PHI2 : return &(map->template get_attribute_container<Orbit::PHI2>());
+			case Orbit::PHI1_PHI2 : return &(map->template get_attribute_container<Orbit::PHI1_PHI2>());
+			case Orbit::PHI1_PHI3 : return &(map->template get_attribute_container<Orbit::PHI1_PHI3>());
+			case Orbit::PHI2_PHI3 : return &(map->template get_attribute_container<Orbit::PHI2_PHI3>());
+			case Orbit::PHI21 : return &(map->template get_attribute_container<Orbit::PHI21>());
+			case Orbit::PHI21_PHI31 : return &(map->template get_attribute_container<Orbit::PHI21_PHI31>());
+			case Orbit::PHI1_PHI2_PHI3 : return &(map->template get_attribute_container<Orbit::PHI1_PHI2_PHI3>());
+			default:
+				return nullptr;
+		}
+	}
```

Comment:
Oui.
Il faut vraiment que j'arrête mes brouillons de PR  ; -)
Je tenais juste à ce qur sur le fond il n'y ait pas de levée de boucliers ! 

Le 25 mai 2016 12:55:37 GMT+02:00, Pierre Kraemer <notifications@github.com> a écrit :
>>  	{
>> -		return chunk_array_->get_type_name();
>> +		switch (orb) {
>> +			case Orbit::DART : return &(map->template
>get_attribute_container<Orbit::DART>());
>> +			case Orbit::PHI1 : return &(map->template
>get_attribute_container<Orbit::PHI1>());
>> +			case Orbit::PHI2 : return &(map->template
>get_attribute_container<Orbit::PHI2>());
>> +			case Orbit::PHI1_PHI2 : return &(map->template
>get_attribute_container<Orbit::PHI1_PHI2>());
>> +			case Orbit::PHI1_PHI3 : return &(map->template
>get_attribute_container<Orbit::PHI1_PHI3>());
>> +			case Orbit::PHI2_PHI3 : return &(map->template
>get_attribute_container<Orbit::PHI2_PHI3>());
>> +			case Orbit::PHI21 : return &(map->template
>get_attribute_container<Orbit::PHI21>());
>> +			case Orbit::PHI21_PHI31 : return &(map->template
>get_attribute_container<Orbit::PHI21_PHI31>());
>> +			case Orbit::PHI1_PHI2_PHI3 : return &(map->template
>get_attribute_container<Orbit::PHI1_PHI2_PHI3>());
>> +			default:
>> +				return nullptr;
>> +		}
>> +	}
>
>Je ne comprends pas ce que cette méthode fait là.
>Elle pourrait être avantageusement remplacée par la méthode suivante
>dans MapBaseData : 
>```c++
>inline const ChunkArrayContainer<uint32>&
>get_const_attribute_container(Orbit orbit) const
>{
>	cgogn_message_assert(orbit < NB_ORBITS, "Unknown orbit parameter");
>	return attributes_[orbit];
>}
>```
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/cgogn/CGoGN_2/pull/208/files/4b965e0c5612c91651f9c2c18d3665a28c5ff99e#r64552634

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.